numpy/core/src/umath/loops_arithmetic.dispatch.c.src

```diff
@@ -20,11 +20,103 @@
+ *     q = TRUNC((n − (−dsign ) + (−nsign))/d) − (−qsign);
  ********************************************************************************/
 #if NPY_SIMD
 /**begin repeat
- *  #sfx = u8, u16, u32, u64#
+ * #sfx    = u8,  u16, u32, u64, s8, s16, s32, s64#
+ * #len    = 8,   16,  32,  64,  8,  16,  32,  64#
+ * #signed = 0*4, 1*4#
  */
+#if @signed@
+static NPY_INLINE void
+simd_divide_by_scalar_contig_@sfx@(char **args, npy_intp len)
+{
+    npyv_@sfx@  a, nsign_d, nsign_a, diff_sign, to_ninf, trunc, floor, neg, vzero;
+    npyv_b@len@ greater_min, noverflow;
+    npy_bool    raise;
+    npy_uint64  tobits;
+
+    npyv_lanetype_@sfx@ *src   = (npyv_lanetype_@sfx@ *) args[0];
+    npyv_lanetype_@sfx@ scalar = *(npyv_lanetype_@sfx@ *) args[1];
+    npyv_lanetype_@sfx@ *dst   = (npyv_lanetype_@sfx@ *) args[2];
+    const int vstep            = npyv_nlanes_@sfx@;
+    const npyv_@sfx@x3 divisor = npyv_divisor_@sfx@(scalar);
+
+    if (NPY_UNLIKELY(-1 == scalar)) {
+        noverflow = npyv_cvt_b@len@_@sfx@(npyv_setall_@sfx@(-1));
+        vzero     = npyv_zero_@sfx@();
+        for (; len >= vstep; len -= vstep, src += vstep, dst += vstep) {
+            a           = npyv_load_@sfx@(src);
+            greater_min = npyv_cmpgt_@sfx@(a, npyv_setall_@sfx@(NPY_MIN_INT@len@));
+            noverflow   = npyv_and_b@len@(noverflow, greater_min);
+            neg         = npyv_ifsub_@sfx@(greater_min, vzero, a, vzero);
+
+            npyv_store_@sfx@(dst, neg);
+        }
+        tobits = npyv_tobits_b@len@(noverflow);
+    #if npyv_nlanes_@sfx@ == 64
+        raise = (~tobits) != 0;
+    #else
+        raise = tobits != (1ULL << vstep)-1;
+    #endif
+
+        for (; len > 0; --len, ++src, ++dst) {
+            npyv_lanetype_@sfx@ a = *src;
+            if (a == NPY_MIN_INT@len@) {
+                raise = NPY_TRUE;
+                *dst  = 0;
+            } else {
+                *dst = -a;
+            }
+        }
+        if (raise) {
+            npy_set_floatstatus_divbyzero();
+        }
+    } else {
+        for (; len >= vstep; len -= vstep, src += vstep, dst += vstep) {
+            nsign_d   = npyv_setall_@sfx@(scalar < 0);
+            a         = npyv_load_@sfx@(src);
+            nsign_a   = npyv_cvt_@sfx@_b@len@(npyv_cmplt_@sfx@(a, nsign_d));
+            nsign_a   = npyv_and_@sfx@(nsign_a, npyv_setall_@sfx@(1));
+            diff_sign = npyv_sub_@sfx@(nsign_a, nsign_d);
+            to_ninf   = npyv_xor_@sfx@(nsign_a, nsign_d);
+            trunc     = npyv_divc_@sfx@(npyv_add_@sfx@(a, diff_sign), divisor);
+            floor     = npyv_sub_@sfx@(trunc, to_ninf);
+
+            npyv_store_@sfx@(dst, floor);
+        }
+
+        for (; len > 0; --len, ++src, ++dst) {
+            const npyv_lanetype_@sfx@ a = *src;
+            if (scalar == 0 || (a == (npyv_lanetype_@sfx@)NPY_MIN_INT@len@ && scalar == (npyv_lanetype_@sfx@)-1)) {
+                 npy_set_floatstatus_divbyzero();
+                 *dst = 0;
+            } else {
```

Comment:
```suggestion

```
overflow already handled by the branch above, and divide by zero handled by the main binary loop